### ISSUE TITLE
There is a problem that mAP is always calculated as 0.0.

### ISSUE THREAD

**Issue by JunhoSer** [Mstfakts/Building-Detection-MaskRCNN#5]
I ran the SpaceNet_train.py file using the weights file you provided. (The learning step is omitted.)
However, unlike the code, random images are not displayed, and mAP is always calculated as 0.0.
Actually, I'm a beginner who just studied deep learning, so I can't figure out which part is wrong.
I am attaching the file(convert txt to py) I ran, so please tell me where the problem is.
Thank you always.

Configurations:
BACKBONE                       resnet50
BACKBONE_STRIDES               [4, 8, 16, 32, 64]
BATCH_SIZE                     1
BBOX_STD_DEV                   [0.1 0.1 0.2 0.2]
COMPUTE_BACKBONE_SHAPE         None
DETECTION_MAX_INSTANCES        350
DETECTION_MIN_CONFIDENCE       0.7
DETECTION_NMS_THRESHOLD        0.3
FPN_CLASSIF_FC_LAYERS_SIZE     1024
GPU_COUNT                      1
GRADIENT_CLIP_NORM             5.0
IMAGES_PER_GPU                 1
IMAGE_CHANNEL_COUNT            3
IMAGE_MAX_DIM                  640
IMAGE_META_SIZE                14
IMAGE_MIN_DIM                  640
IMAGE_MIN_SCALE                0
IMAGE_RESIZE_MODE              square
IMAGE_SHAPE                    [640 640   3]
LEARNING_MOMENTUM              0.9
LEARNING_RATE                  0.001
LOSS_WEIGHTS                   {'rpn_class_loss': 1.0, 'rpn_bbox_loss': 1.0, 'mrcnn_class_loss': 1.0, 'mrcnn_bbox_loss': 1.0, 'mrcnn_mask_loss': 1.0}
MASK_POOL_SIZE                 14
MASK_SHAPE                     [28, 28]
MAX_GT_INSTANCES               250
MEAN_PIXEL                     [123.7 116.8 103.9]
MINI_MASK_SHAPE                (56, 56)
NAME                           SpaceNet
NUM_CLASSES                    2
POOL_SIZE                      7
POST_NMS_ROIS_INFERENCE        1000
POST_NMS_ROIS_TRAINING         2000
PRE_NMS_LIMIT                  6000
ROI_POSITIVE_RATIO             0.33
RPN_ANCHOR_RATIOS              [0.25, 1, 4]
RPN_ANCHOR_SCALES              (8, 16, 32, 64, 128)
RPN_ANCHOR_STRIDE              1
RPN_BBOX_STD_DEV               [0.1 0.1 0.2 0.2]
RPN_NMS_THRESHOLD              0.7
RPN_TRAIN_ANCHORS_PER_IMAGE    256
STEPS_PER_EPOCH                500
TOP_DOWN_PYRAMID_SIZE          256
TRAIN_BN                       False
TRAIN_ROIS_PER_IMAGE           32
USE_MINI_MASK                  True
USE_RPN_ROIS                   True
VALIDATION_STEPS               50
WEIGHT_DECAY                   0.0001

1165
C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\mask_rcnn-2.1-py3.6.egg\mrcnn\visualize.py:56: UserWarning:

Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.

1336
788
737
WARNING:tensorflow:From C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be
 removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\keras\backend\tensorflow_backend.py:1154: calling reduce_max_v1 (from tensorflow.python.ops.math_ops) with keep_dims is deprecate
d and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
WARNING:tensorflow:From C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\keras\backend\tensorflow_backend.py:1188: calling reduce_sum_v1 (from tensorflow.python.ops.math_ops) with keep_dims is deprecate
d and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
WARNING:tensorflow:From C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\mask_rcnn-2.1-py3.6.egg\mrcnn\model.py:772: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a
 future version.
Instructions for updating:
Use tf.cast instead.
Loading weights from  D:\mask_rcnn_spacenet_0151.h5
2021-11-09 01:06:42.125108: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2
2021-11-09 01:06:42.267542: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: 
name: GeForce RTX 3060 Ti major: 8 minor: 6 memoryClockRate(GHz): 1.695
pciBusID: 0000:01:00.0
totalMemory: 8.00GiB freeMemory: 6.99GiB
2021-11-09 01:06:42.268127: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0
2021-11-09 01:10:02.188573: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-11-09 01:10:02.188862: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990]      0 
2021-11-09 01:10:02.189157: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0:   N
2021-11-09 01:10:02.189519: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 6714 MB memory) -> physical GPU (device: 0,
 name: GeForce RTX 3060 Ti, pci bus id: 0000:01:00.0, compute capability: 8.6)
original_image           shape: (640, 640, 3)         min:    6.00000  max:  250.00000  uint8
image_meta               shape: (14,)                 min:    0.00000  max:  650.00000  float64
gt_class_id              shape: (0,)                  min:             max:             int32
gt_bbox                  shape: (0, 4)                min:             max:             int32
gt_mask                  shape: (640, 640, 0)         min:             max:             bool

*** No instances to display ***

C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\mask_rcnn-2.1-py3.6.egg\mrcnn\visualize.py:167: UserWarning:

Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.

Processing 1 images
image                    shape: (640, 640, 3)         min:    6.00000  max:  250.00000  uint8
molded_images            shape: (1, 640, 640, 3)      min: -117.70000  max:  133.20000  float64
image_metas              shape: (1, 14)               min:    0.00000  max:  640.00000  int32
anchors                  shape: (1, 102300, 4)        min:   -0.20031  max:    1.10016  float32
2021-11-09 01:10:04.638200: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library cublas64_100.dll locally

*** No instances to display ***

mAP:  0.
[SpaceNet_train_modified.txt](https://github.com/Mstfakts/Building-Detection-MaskRCNN/files/7526264/SpaceNet_train_modified.txt)
0

**Comment by JunhoSeo2075** [comment]
![aaa](https://user-images.githubusercontent.com/65344591/141613138-529fc466-3711-421b-90ba-d7671bb3ab02.png)
My guess is that Mask does not apply properly to the image. However, since I correctly generated Mask as a png file, I don't know why the problem occurred.

**Comment by JunhoSer** [comment]
I've solved a lot of problems, but there's still a detection problem. I applied the weight file you sent me, but my model does not detect the building properly. Also, the mAP is still 0.0.
I attach the file I turned (ipynb format) in .txt
Help me plz :(
![wtf](https://user-images.githubusercontent.com/65344591/142000680-908e111e-1632-4306-bd0d-050ebbdd3c66.png
[Training.txt](https://github.com/Mstfakts/Building-Detection-MaskRCNN/files/7546966/Training.txt)
)

**Comment by JunhoSer** [comment]
> I ran the SpaceNet_train.py file using the weights file you provided. (The learning step is omitted.) However, unlike the code, random images are not displayed, and mAP is always calculated as 0.0. Actually, I'm a beginner who just studied deep learning, so I can't figure out which part is wrong. I am attaching the file(convert txt to py) I ran, so please tell me where the problem is. Thank you always.
> 
> Configurations: BACKBONE resnet50 BACKBONE_STRIDES [4, 8, 16, 32, 64] BATCH_SIZE 1 BBOX_STD_DEV [0.1 0.1 0.2 0.2] COMPUTE_BACKBONE_SHAPE None DETECTION_MAX_INSTANCES 350 DETECTION_MIN_CONFIDENCE 0.7 DETECTION_NMS_THRESHOLD 0.3 FPN_CLASSIF_FC_LAYERS_SIZE 1024 GPU_COUNT 1 GRADIENT_CLIP_NORM 5.0 IMAGES_PER_GPU 1 IMAGE_CHANNEL_COUNT 3 IMAGE_MAX_DIM 640 IMAGE_META_SIZE 14 IMAGE_MIN_DIM 640 IMAGE_MIN_SCALE 0 IMAGE_RESIZE_MODE square IMAGE_SHAPE [640 640 3] LEARNING_MOMENTUM 0.9 LEARNING_RATE 0.001 LOSS_WEIGHTS {'rpn_class_loss': 1.0, 'rpn_bbox_loss': 1.0, 'mrcnn_class_loss': 1.0, 'mrcnn_bbox_loss': 1.0, 'mrcnn_mask_loss': 1.0} MASK_POOL_SIZE 14 MASK_SHAPE [28, 28] MAX_GT_INSTANCES 250 MEAN_PIXEL [123.7 116.8 103.9] MINI_MASK_SHAPE (56, 56) NAME SpaceNet NUM_CLASSES 2 POOL_SIZE 7 POST_NMS_ROIS_INFERENCE 1000 POST_NMS_ROIS_TRAINING 2000 PRE_NMS_LIMIT 6000 ROI_POSITIVE_RATIO 0.33 RPN_ANCHOR_RATIOS [0.25, 1, 4] RPN_ANCHOR_SCALES (8, 16, 32, 64, 128) RPN_ANCHOR_STRIDE 1 RPN_BBOX_STD_DEV [0.1 0.1 0.2 0.2] RPN_NMS_THRESHOLD 0.7 RPN_TRAIN_ANCHORS_PER_IMAGE 256 STEPS_PER_EPOCH 500 TOP_DOWN_PYRAMID_SIZE 256 TRAIN_BN False TRAIN_ROIS_PER_IMAGE 32 USE_MINI_MASK True USE_RPN_ROIS True VALIDATION_STEPS 50 WEIGHT_DECAY 0.0001
> 
> 1165 C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\mask_rcnn-2.1-py3.6.egg\mrcnn\visualize.py:56: UserWarning:
> 
> Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
> 
> 1336 788 737 WARNING:tensorflow:From C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\tensorflow\python\framework\op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version. Instructions for updating: Colocations handled automatically by placer. WARNING:tensorflow:From C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\keras\backend\tensorflow_backend.py:1154: calling reduce_max_v1 (from tensorflow.python.ops.math_ops) with keep_dims is deprecate d and will be removed in a future version. Instructions for updating: keep_dims is deprecated, use keepdims instead WARNING:tensorflow:From C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\keras\backend\tensorflow_backend.py:1188: calling reduce_sum_v1 (from tensorflow.python.ops.math_ops) with keep_dims is deprecate d and will be removed in a future version. Instructions for updating: keep_dims is deprecated, use keepdims instead WARNING:tensorflow:From C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\mask_rcnn-2.1-py3.6.egg\mrcnn\model.py:772: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version. Instructions for updating: Use tf.cast instead. Loading weights from D:\mask_rcnn_spacenet_0151.h5 2021-11-09 01:06:42.125108: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 2021-11-09 01:06:42.267542: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1433] Found device 0 with properties: name: GeForce RTX 3060 Ti major: 8 minor: 6 memoryClockRate(GHz): 1.695 pciBusID: 0000:01:00.0 totalMemory: 8.00GiB freeMemory: 6.99GiB 2021-11-09 01:06:42.268127: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1512] Adding visible gpu devices: 0 2021-11-09 01:10:02.188573: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] Device interconnect StreamExecutor with strength 1 edge matrix: 2021-11-09 01:10:02.188862: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0 2021-11-09 01:10:02.189157: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1003] 0: N 2021-11-09 01:10:02.189519: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1115] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 6714 MB memory) -> physical GPU (device: 0, name: GeForce RTX 3060 Ti, pci bus id: 0000:01:00.0, compute capability: 8.6) original_image shape: (640, 640, 3) min: 6.00000 max: 250.00000 uint8 image_meta shape: (14,) min: 0.00000 max: 650.00000 float64 gt_class_id shape: (0,) min: max: int32 gt_bbox shape: (0, 4) min: max: int32 gt_mask shape: (640, 640, 0) min: max: bool
> 
> *** No instances to display ***
> 
> C:\Users\Seo\anaconda3\envs\mrcnn\lib\site-packages\mask_rcnn-2.1-py3.6.egg\mrcnn\visualize.py:167: UserWarning:
> 
> Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
> 
> Processing 1 images image shape: (640, 640, 3) min: 6.00000 max: 250.00000 uint8 molded_images shape: (1, 640, 640, 3) min: -117.70000 max: 133.20000 float64 image_metas shape: (1, 14) min: 0.00000 max: 640.00000 int32 anchors shape: (1, 102300, 4) min: -0.20031 max: 1.10016 float32 2021-11-09 01:10:04.638200: I tensorflow/stream_executor/dso_loader.cc:152] successfully opened CUDA library cublas64_100.dll locally
> 
> *** No instances to display ***
> 
> mAP: 0. [SpaceNet_train_modified.txt](https://github.com/Mstfakts/Building-Detection-MaskRCNN/files/7526264/SpaceNet_train_modified.txt) 0
